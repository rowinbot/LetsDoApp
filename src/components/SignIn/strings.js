export default {
  titleMain: "LET'S DO",
  buttonSignIn: "SIGN IN"
}