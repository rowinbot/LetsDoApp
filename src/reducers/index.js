import { combineReducers } from 'redux';
import routes from './routes';
import colors from './colors';

export default combineReducers({
  routes,
  colors
});