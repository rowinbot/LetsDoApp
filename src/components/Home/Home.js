import React, { Component } from 'react';

import { 
  Text,
  TextInput,
  View,
  TouchableHighlight
} from 'react-native';

import { Actions } from 'react-native-router-flux'
import PropTypes from 'prop-types';
import LoginInput from '../../controls/LoginInput'
import GradientScrollView from '../../controls/GradientScrollView'
import TopBar from '../../controls/TopBar'

import { alphaHex } from '../../helpers/Styles'

import styles from './styles';
import strings from './strings'

export default class SignIn extends Component<{}> {
  constructor(props) {
    super(props);
    
    this.state = {
      user: {
        userName: "",
        password: ""
      }
    }
  }
  
  static propTypes = {
    ui: PropTypes.object.isRequired
  };
  
  render() {   
    const ui = this.props.ui;
               
    return (
      <GradientScrollView>
        <TopBar title={strings.titleMain} />
      </GradientScrollView>
    );
  }
}