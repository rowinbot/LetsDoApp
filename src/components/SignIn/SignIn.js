import React, { Component } from 'react';

import { 
  Text,
  TextInput,
  View,
  ScrollView,
  TouchableHighlight
} from 'react-native';

import { Actions } from 'react-native-router-flux'
import LoginInput from '../../controls/LoginInput'
import BackgroundButton from '../../controls/buttons/BackgroundButton'
import PropTypes from 'prop-types';

import styles from './styles';
import strings from './strings'

export default class SignIn extends Component<{}> {
  constructor(props) {
    super(props);
    
    this.state = {
      user: {
        userName: "",
        password: ""
      }
    }
  }
  
  static propTypes = {
    ui: PropTypes.object.isRequired
  };
  
  render() {   
    const ui = this.props.ui;
               
    return (
      <ScrollView 
        style={ui.colorBackgroud}
        contentContainerStyle={styles.containerScrollView}
        keyboardShouldPersistTaps={"handled"}
        >
        <Text style={[styles.titleMain, ui.colorText]}>
          {strings.titleMain}
        </Text>
        
        <LoginInput
          onChange={e => this.setState({user: { ...this.state.user, userName: e.nativeEvent.text }})}
          placeholder="USER NAME"
          value={this.state.user.userName}/>
            
        <LoginInput
          onChange={e => this.setState({user: { ...this.state.user, password: e.nativeEvent.text }})}
          placeholder="PASSWORD"
          value={this.state.user.password} 
          secureTextEntry={true}/>

        <BackgroundButton
          handler={ () => {  } }
          text={strings.buttonSignIn}/>
      </ScrollView>
    );
  }
}