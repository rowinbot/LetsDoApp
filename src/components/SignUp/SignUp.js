import React, { Component } from 'react';
import {
  Text,
  TextInput,
  View,
  ScrollView,
  TouchableOpacity
} from 'react-native';

import { Actions } from 'react-native-router-flux'

import LinearGradient from 'react-native-linear-gradient';

import LoginInput from '../../controls/LoginInput'
import BorderedButton from '../../controls/buttons/BorderedButton'
import BackgroundButton from '../../controls/buttons/BackgroundButton'
import UserPicker from '../../controls/UserPicker'
import PropTypes from 'prop-types';

import styles from './styles';
import strings from './strings'

export default class SignUp extends Component<{}> {
  constructor(props) {
    super(props);
    
    this.state = {
      user: {
        name: "",
        userName: "",
        email: "",
        password: ""
      }
    }
  }
  
  static propTypes = {
    ui: PropTypes.object.isRequired,
    onChangeUserType: PropTypes.func.isRequired
  };
  
  render() {
    const changeUserType = this.props.onChangeUserType,
          ui             = this.props.ui;
                        
    return (
      <ScrollView 
        style={ui.colorBackgroud}
        contentContainerStyle={styles.containerScrollView}
        keyboardShouldPersistTaps={"handled"}
        >
        <Text style={[styles.titleMain, ui.colorText]}>
          {strings.titleMain}
        </Text>
        
        <UserPicker 
          handler={
            (number) => changeUserType(number)
          } />
        
        <BorderedButton
          handler={ (e) => { Actions.signIn() } }
          textSize={20}
          text={strings.buttonSignIn}/>
        
        <LoginInput
          onChange={e => this.setState({user: { ...this.state.user, name: e.nativeEvent.text }})}
          placeholder="NAME"
          value={this.state.user.name}/>
          
        <LoginInput
          onChange={e => this.setState({user: { ...this.state.user, userName: e.nativeEvent.text }})}
          placeholder="USER NAME"
          value={this.state.user.userName}/>
            
        <LoginInput
          onChange={e => this.setState({user: { ...this.state.user, email: e.nativeEvent.text }})}
          placeholder="EMAIL"
          value={this.state.user.email}/>
          
        <LoginInput
          onChange={e => this.setState({user: { ...this.state.user, password: e.nativeEvent.text }})}
          placeholder="PASSWORD"
          value={this.state.user.password} 
          secureTextEntry={true}/>

        <BackgroundButton
          handler={ (e) => { Actions.home() } }
          text={strings.buttonSignUp}/>
      </ScrollView>
    );
  }
}