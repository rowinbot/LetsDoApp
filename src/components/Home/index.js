import Home from './Home'
import { connect } from 'react-redux';

import withBackHandler from '../backHandler';

function mapStateToProps(state) {
  return {
    ui: state.colors.ui
  };
}

export default connect(
  mapStateToProps
)(withBackHandler(Home));