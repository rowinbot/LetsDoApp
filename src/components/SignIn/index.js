import SignIn from './SignIn'
import { connect } from 'react-redux';

import withBackHandler from '../backHandler';

function mapStateToProps(state) {
  return {
    ui: state.colors.ui
  };
}

export default connect(
  mapStateToProps
)(withBackHandler(SignIn));