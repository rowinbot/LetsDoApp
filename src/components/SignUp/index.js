import SignUp from './SignUp'
import { connect } from 'react-redux';
import { changeColor } from '../../actions/colors'

function mapStateToProps(state) {
  return {
    ui: state.colors.ui,
    routes: state.routes
  };
}

const mapDispatchToProps = dispatch => {
  return {
    onChangeUserType: color => {
      dispatch(changeColor(color))
    }
  }
}


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignUp);