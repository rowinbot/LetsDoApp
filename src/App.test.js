import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
/*const App = require('../App')*/

import { Actions, Scene, Router } from 'react-native-router-flux'
import { connect, Provider } from 'react-redux'
import { createStore, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'
import SplashScreen from 'react-native-splash-screen'

jest.mock('react-native-splash-screen')

import SignUp from '../src/components/SignUp'
import SignIn from '../src/components/SignIn'
import Home from '../src/components/Home'

const RouterWithRedux = connect()(Router)
import reducers from '../src/reducers'

const store = compose(
  applyMiddleware(thunk)
)(createStore)(reducers)

const SignUpScene = connect()(SignUp)
const SignInScene = connect()(SignIn)
const HomeScene = connect()(Home)

const scenes = Actions.create(
  <Scene key="root">
    <Scene key="signUp" component={SignUpScene} title="SignUp" initial={true} hideNavBar={true}/>
    <Scene key="signIn" component={SignInScene} title="SignIn" hideNavBar={true}/>
    <Scene key="home" component={HomeScene} title="Home" hideNavBar={true}/>
  </Scene>
);

class App extends React.Component<{}> {
  componentDidMount() {
    // do stuff while splash screen is shown
    // After having done stuff (such as async tasks) hide the splash screen
    SplashScreen.hide()
  }

  render() {
    return (
      <Provider store={store}>
        <RouterWithRedux scenes={scenes}/>
      </Provider>
    );
  }
}

describe('App', () => {
  it('renders correctly', () => {
    const instance = renderer.create(<App/>).toJSON();

    expect(instance).toMatchSnapshot();
  });
});