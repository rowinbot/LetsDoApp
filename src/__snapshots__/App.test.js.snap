// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "alignItems": "center",
              "flexGrow": 1,
            }
          }
          keyboardShouldPersistTaps="handled"
          style={
            Object {
              "backgroundColor": "#273138",
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Montserrat-Medium",
                    "fontSize": 30,
                    "marginTop": 40,
                  },
                  Object {
                    "color": "#F4565C",
                  },
                ]
              }
            >
              Who Are You
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "marginTop": 8,
                  "width": 250,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#FC5659",
                    "borderColor": "#F4565C",
                    "borderRadius": 50,
                    "borderWidth": 1,
                    "height": 65,
                    "opacity": 1,
                    "width": 65,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#B664E3",
                    "borderColor": "#F4565C",
                    "borderRadius": 50,
                    "borderWidth": 1,
                    "height": 65,
                    "opacity": 1,
                    "width": 65,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#F4565C",
                    "borderRadius": 50,
                    "borderWidth": 1,
                    "height": 65,
                    "opacity": 1,
                    "width": 65,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 60,
                  "justifyContent": "center",
                  "marginTop": 20,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "rgba(53,62,67,0.4)",
                    "borderRadius": 25,
                    "borderWidth": 4,
                    "height": 40,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Montserrat-Medium",
                      },
                      Object {
                        "color": "#F4565C",
                      },
                      Object {
                        "fontSize": 20,
                      },
                    ]
                  }
                >
                  Already have an account?
                </Text>
              </View>
            </View>
            <TextInput
              allowFontScaling={true}
              colors={
                Object {
                  "colorAccent": "#F4565C",
                  "colorBackgroud": Object {
                    "backgroundColor": "#273138",
                  },
                  "colorBorder": Object {
                    "borderColor": "#F4565C",
                  },
                  "colorButtonBorder": Object {
                    "borderColor": "#353E43",
                  },
                  "colorButtons": Object {
                    "backgroundColor": "#353E43",
                    "borderColor": "#353E43",
                  },
                  "colorDark": "#273138",
                  "colorPrimary": "#353E43",
                  "colorText": Object {
                    "color": "#F4565C",
                  },
                }
              }
              disableFullscreenUI={true}
              dispatch={[Function]}
              onChange={[Function]}
              placeholder="NAME"
              placeholderTextColor="rgba(244,86,92,0.5)"
              returnKeyType="next"
              selectTextOnFocus={false}
              selectionColor="rgba(244,86,92,0.2)"
              style={
                Array [
                  Object {
                    "borderRadius": 50,
                    "borderWidth": 2,
                    "fontSize": 17,
                    "height": 40,
                    "marginTop": 10,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "textDecorationLine": "none",
                    "width": 250,
                  },
                  Object {
                    "color": "#F4565C",
                  },
                  Object {
                    "borderColor": "rgba(53,62,67,0.4)",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <TextInput
              allowFontScaling={true}
              colors={
                Object {
                  "colorAccent": "#F4565C",
                  "colorBackgroud": Object {
                    "backgroundColor": "#273138",
                  },
                  "colorBorder": Object {
                    "borderColor": "#F4565C",
                  },
                  "colorButtonBorder": Object {
                    "borderColor": "#353E43",
                  },
                  "colorButtons": Object {
                    "backgroundColor": "#353E43",
                    "borderColor": "#353E43",
                  },
                  "colorDark": "#273138",
                  "colorPrimary": "#353E43",
                  "colorText": Object {
                    "color": "#F4565C",
                  },
                }
              }
              disableFullscreenUI={true}
              dispatch={[Function]}
              onChange={[Function]}
              placeholder="USER NAME"
              placeholderTextColor="rgba(244,86,92,0.5)"
              returnKeyType="next"
              selectTextOnFocus={false}
              selectionColor="rgba(244,86,92,0.2)"
              style={
                Array [
                  Object {
                    "borderRadius": 50,
                    "borderWidth": 2,
                    "fontSize": 17,
                    "height": 40,
                    "marginTop": 10,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "textDecorationLine": "none",
                    "width": 250,
                  },
                  Object {
                    "color": "#F4565C",
                  },
                  Object {
                    "borderColor": "rgba(53,62,67,0.4)",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <TextInput
              allowFontScaling={true}
              colors={
                Object {
                  "colorAccent": "#F4565C",
                  "colorBackgroud": Object {
                    "backgroundColor": "#273138",
                  },
                  "colorBorder": Object {
                    "borderColor": "#F4565C",
                  },
                  "colorButtonBorder": Object {
                    "borderColor": "#353E43",
                  },
                  "colorButtons": Object {
                    "backgroundColor": "#353E43",
                    "borderColor": "#353E43",
                  },
                  "colorDark": "#273138",
                  "colorPrimary": "#353E43",
                  "colorText": Object {
                    "color": "#F4565C",
                  },
                }
              }
              disableFullscreenUI={true}
              dispatch={[Function]}
              onChange={[Function]}
              placeholder="EMAIL"
              placeholderTextColor="rgba(244,86,92,0.5)"
              returnKeyType="next"
              selectTextOnFocus={false}
              selectionColor="rgba(244,86,92,0.2)"
              style={
                Array [
                  Object {
                    "borderRadius": 50,
                    "borderWidth": 2,
                    "fontSize": 17,
                    "height": 40,
                    "marginTop": 10,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "textDecorationLine": "none",
                    "width": 250,
                  },
                  Object {
                    "color": "#F4565C",
                  },
                  Object {
                    "borderColor": "rgba(53,62,67,0.4)",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <TextInput
              allowFontScaling={true}
              colors={
                Object {
                  "colorAccent": "#F4565C",
                  "colorBackgroud": Object {
                    "backgroundColor": "#273138",
                  },
                  "colorBorder": Object {
                    "borderColor": "#F4565C",
                  },
                  "colorButtonBorder": Object {
                    "borderColor": "#353E43",
                  },
                  "colorButtons": Object {
                    "backgroundColor": "#353E43",
                    "borderColor": "#353E43",
                  },
                  "colorDark": "#273138",
                  "colorPrimary": "#353E43",
                  "colorText": Object {
                    "color": "#F4565C",
                  },
                }
              }
              disableFullscreenUI={true}
              dispatch={[Function]}
              onChange={[Function]}
              placeholder="PASSWORD"
              placeholderTextColor="rgba(244,86,92,0.5)"
              returnKeyType="next"
              secureTextEntry={true}
              selectTextOnFocus={false}
              selectionColor="rgba(244,86,92,0.2)"
              style={
                Array [
                  Object {
                    "borderRadius": 50,
                    "borderWidth": 2,
                    "fontSize": 17,
                    "height": 40,
                    "marginTop": 10,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "textDecorationLine": "none",
                    "width": 250,
                  },
                  Object {
                    "color": "#F4565C",
                  },
                  Object {
                    "borderColor": "rgba(53,62,67,0.4)",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 60,
                  "justifyContent": "center",
                  "marginTop": 20,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "rgba(53,62,67,0.4)",
                    "borderRadius": 25,
                    "borderWidth": 0,
                    "height": 40,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Montserrat-Medium",
                      },
                      Object {
                        "color": "#F4565C",
                      },
                      Object {
                        "fontSize": 26,
                      },
                    ]
                  }
                >
                  SIGN UP
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
